src/core/Song.cpp

```diff
@@ -339,14 +338,17 @@ void Song::processNextBuffer()
 				// if looping-mode is enabled and we have got
 				// past the looping range, return to the 
 				// beginning of the range
+				if( m_playPos[m_playMode] == tl->loopEnd() - 1 )
+				{
+					emit updateSampleTracks();
+				}
 				if( m_playPos[m_playMode] >= tl->loopEnd() )
```

Comment:
I think you should just try and move the whole loop one tick earlier.

```
if( m_playPos[m_playMode] >= tl->loopEnd() )
{
     m_playPos[m_playMode].setTicks( tl->loopBegin().getTicks() );
```

to something like (pseudocodeish...):

```
if( m_playPos[m_playMode] >= tl->loopEnd().getTicks() - 1 )
{
     m_playPos[m_playMode].setTicks( tl->loopBegin().getTicks() - 1 );
```